// @wix/identity_roles-management/build/es/src/identity-v1-contributor-roles-management.types.js
export var ResourceType;
(function (ResourceType) {
    ResourceType["UNKNOWN_RESOURCE_TYPE"] = "UNKNOWN_RESOURCE_TYPE";
    ResourceType["SITE"] = "SITE";
})(ResourceType || (ResourceType = {}));
export var ConditionAttributeType;
(function (ConditionAttributeType) {
    ConditionAttributeType["UNKNOWN_CONDITION_TYPE"] = "UNKNOWN_CONDITION_TYPE";
    ConditionAttributeType["FOLDER"] = "FOLDER";
})(ConditionAttributeType || (ConditionAttributeType = {}));
export var SubjectType;
(function (SubjectType) {
    SubjectType["UNKNOWN"] = "UNKNOWN";
    SubjectType["ACCOUNT"] = "ACCOUNT";
    SubjectType["USER"] = "USER";
    SubjectType["USER_GROUP"] = "USER_GROUP";
    SubjectType["MEMBER_GROUP"] = "MEMBER_GROUP";
    SubjectType["VISITOR_GROUP"] = "VISITOR_GROUP";
    SubjectType["EXTERNAL_APP"] = "EXTERNAL_APP";
    SubjectType["ACCOUNT_GROUP"] = "ACCOUNT_GROUP";
    SubjectType["WIX_APP"] = "WIX_APP";
})(SubjectType || (SubjectType = {}));
export var SubjectContextType;
(function (SubjectContextType) {
    SubjectContextType["UNKNOWN_CTX"] = "UNKNOWN_CTX";
    SubjectContextType["ORG_CTX"] = "ORG_CTX";
    SubjectContextType["ACCOUNT_CTX"] = "ACCOUNT_CTX";
})(SubjectContextType || (SubjectContextType = {}));
/** Invite status stating whether the invite was accepted, waiting to be accepted, deleted etc.. */
export var InviteStatus;
(function (InviteStatus) {
    InviteStatus["Pending"] = "Pending";
    InviteStatus["Used"] = "Used";
    InviteStatus["Deleted"] = "Deleted";
    InviteStatus["Declined"] = "Declined";
    InviteStatus["Expired"] = "Expired";
})(InviteStatus || (InviteStatus = {}));
/** Enum to represent different types of contributors quota. */
export var Type;
(function (Type) {
    Type["UNKNOWN"] = "UNKNOWN";
    Type["LIMITED"] = "LIMITED";
    Type["UNLIMITED"] = "UNLIMITED";
})(Type || (Type = {}));
export var FieldSet;
(function (FieldSet) {
    FieldSet["UNKNOWN"] = "UNKNOWN";
    /** Include only `account_id` and `account_owner_id` fields. */
    FieldSet["META_DATA"] = "META_DATA";
})(FieldSet || (FieldSet = {}));
