// @wix/ecom_tip-settings/build/es/src/tips-settings-v1-tip-settings-tip-settings.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformSDKFloatToRESTFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformSDKFieldMaskToRESTFieldMask } from '@wix/sdk-runtime/transformations/field-mask';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixTipsSettingsV1TipSettingsServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/tip-settings',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/tip-settings',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/_api/tip-settings',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/tip-settings',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/ecom_tip-settings';
/**
 * Retrieves a list of tip of up to 100 tip settings, given the provided paging, filtering, and sorting.
 *
 * Query Tip Settings runs with these defaults, which you can override:
 * - `cursorPaging.limit` is `50`.
 * - Sorted by `id` in ascending order.
 *
 * For field support for filters and sorting, see
 * [Tip Settings: Supported Filters and Sorting](https://dev.wix.com/docs/rest/business-solutions/e-commerce/tips/tip-settings/supported-filters).
 *
 * Use [UTC time](https://en.wikipedia.org/wiki/Coordinated_Universal_Time)
 * when setting a filter for dates.
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/api/rest/getting-started/api-query-language),
 * [Sorting and Paging](https://dev.wix.com/api/rest/getting-started/sorting-and-paging),
 * and [Field Projection](https://dev.wix.com/api/rest/getting-started/field-projection).
 */
export function queryTipSettings(payload) {
    function __queryTipSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.tips.settings.v1.tip_settings',
            method: 'POST',
            methodFqn: 'wix.tips.settings.v1.TipSettingsService.QueryTipSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsSettingsV1TipSettingsServiceUrl({
                protoPath: '/v1/tip-settings/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'tipSettings.createdDate' },
                        { path: 'tipSettings.updatedDate' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [{ path: 'tipSettings.presets.value' }],
                },
            ]),
        };
        return metadata;
    }
    return __queryTipSettings;
}
/**
 * Updates a `tipSettings` object.
 *
 *
 * You can't update the `default` field.
 */
export function updateTipSettings(payload) {
    function __updateTipSettings({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'tipSettings.createdDate' },
                    { path: 'tipSettings.updatedDate' },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [{ path: 'tipSettings.presets.value' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.tips.settings.v1.tip_settings',
            method: 'PATCH',
            methodFqn: 'wix.tips.settings.v1.TipSettingsService.UpdateTipSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsSettingsV1TipSettingsServiceUrl({
                protoPath: '/v1/tip-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'tipSettings.createdDate' },
                        { path: 'tipSettings.updatedDate' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [{ path: 'tipSettings.presets.value' }],
                },
            ]),
        };
        return metadata;
    }
    return __updateTipSettings;
}
/**
 * Creates a `tipSettings` object.
 *
 *
 * You can't create a `tipSettings` object with `{"default": true}`.
 */
export function createTipSettings(payload) {
    function __createTipSettings({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'tipSettings.createdDate' },
                    { path: 'tipSettings.updatedDate' },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [{ path: 'tipSettings.presets.value' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.tips.settings.v1.tip_settings',
            method: 'POST',
            methodFqn: 'wix.tips.settings.v1.TipSettingsService.CreateTipSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsSettingsV1TipSettingsServiceUrl({
                protoPath: '/v1/tip-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'tipSettings.createdDate' },
                        { path: 'tipSettings.updatedDate' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [{ path: 'tipSettings.presets.value' }],
                },
            ]),
        };
        return metadata;
    }
    return __createTipSettings;
}
/**
 * Creates a default `tipSettings` object.
 *
 *
 * The call fails if there is already an existing default `tipSettings` object.
 */
export function createDefaultTipSettings(payload) {
    function __createDefaultTipSettings({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'tipSettings.createdDate' },
                    { path: 'tipSettings.updatedDate' },
                ],
            },
            {
                transformFn: transformSDKFloatToRESTFloat,
                paths: [{ path: 'tipSettings.presets.value' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.tips.settings.v1.tip_settings',
            method: 'POST',
            methodFqn: 'wix.tips.settings.v1.TipSettingsService.CreateDefaultTipSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsSettingsV1TipSettingsServiceUrl({
                protoPath: '/v1/tip-settings/create-default',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'tipSettings.createdDate' },
                        { path: 'tipSettings.updatedDate' },
                    ],
                },
                {
                    transformFn: transformRESTFloatToSDKFloat,
                    paths: [{ path: 'tipSettings.presets.value' }],
                },
            ]),
        };
        return metadata;
    }
    return __createDefaultTipSettings;
}
/**
 * Deletes a `tipSettings` object.
 *
 *
 * You can't delete the default `tipSettings` object.
 */
export function deleteTipSettings(payload) {
    function __deleteTipSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.tips.settings.v1.tip_settings',
            method: 'DELETE',
            methodFqn: 'wix.tips.settings.v1.TipSettingsService.DeleteTipSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixTipsSettingsV1TipSettingsServiceUrl({
                protoPath: '/v1/tip-settings/{tipSettingsId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __deleteTipSettings;
}
