// @wix/ecom_orders/build/es/src/ecom-v1-order-orders.context.js
import { preparePaymentCollection as publicPreparePaymentCollection, getPaymentCollectabilityStatus as publicGetPaymentCollectabilityStatus, voidAuthorizedPayments as publicVoidAuthorizedPayments, captureAuthorizedPayments as publicCaptureAuthorizedPayments, getOrder as publicGetOrder, searchOrders as publicSearchOrders, createOrder as publicCreateOrder, updateOrder as publicUpdateOrder, bulkUpdateOrders as publicBulkUpdateOrders, cancelOrder as publicCancelOrder, bulkUpdateOrderTags as publicBulkUpdateOrderTags, } from './ecom-v1-order-orders.public';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onOrderPaymentStatusUpdated as publicOnOrderPaymentStatusUpdated } from './ecom-v1-order-orders.public';
import { onOrderUpdated as publicOnOrderUpdated } from './ecom-v1-order-orders.public';
import { onOrderCreated as publicOnOrderCreated } from './ecom-v1-order-orders.public';
import { onOrderCanceled as publicOnOrderCanceled } from './ecom-v1-order-orders.public';
import { onOrderApproved as publicOnOrderApproved } from './ecom-v1-order-orders.public';
export const preparePaymentCollection = /*#__PURE__*/ createRESTModule(publicPreparePaymentCollection);
export const getPaymentCollectabilityStatus = /*#__PURE__*/ createRESTModule(publicGetPaymentCollectabilityStatus);
export const voidAuthorizedPayments = /*#__PURE__*/ createRESTModule(publicVoidAuthorizedPayments);
export const captureAuthorizedPayments = /*#__PURE__*/ createRESTModule(publicCaptureAuthorizedPayments);
export const getOrder = /*#__PURE__*/ createRESTModule(publicGetOrder);
export const searchOrders = /*#__PURE__*/ createRESTModule(publicSearchOrders);
export const createOrder = /*#__PURE__*/ createRESTModule(publicCreateOrder);
export const updateOrder = /*#__PURE__*/ createRESTModule(publicUpdateOrder);
export const bulkUpdateOrders = /*#__PURE__*/ createRESTModule(publicBulkUpdateOrders);
export const cancelOrder = /*#__PURE__*/ createRESTModule(publicCancelOrder);
export const bulkUpdateOrderTags = /*#__PURE__*/ createRESTModule(publicBulkUpdateOrderTags);
export { publicOnOrderPaymentStatusUpdated };
/**
 * Triggered when an order's payment status is updated to `"PAID"`.
 */
export const onOrderPaymentStatusUpdated = createEventModule(publicOnOrderPaymentStatusUpdated);
export { publicOnOrderUpdated };
/** */
export const onOrderUpdated = createEventModule(publicOnOrderUpdated);
export { publicOnOrderCreated };
/**
 * Triggered when an order is created.
 * Learn more about [webhook payload structure](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/orders/order-object-conversion#webhook-conversion-table).
 */
export const onOrderCreated = createEventModule(publicOnOrderCreated);
export { publicOnOrderCanceled };
/**
 * Triggered when an order is canceled.
 * Learn more about [eCommerce webhook payload structure](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/orders/order-object-conversion#webhook-conversion-table).
 */
export const onOrderCanceled = createEventModule(publicOnOrderCanceled);
export { publicOnOrderApproved };
/**
 * Triggered when an order is created and its status is updated to `"APPROVED"`.
 * Learn more about [eCommerce webhook payload structure](https://dev.wix.com/docs/rest/api-reference/wix-e-commerce/orders/order-object-conversion#webhook-conversion-table).
 */
export const onOrderApproved = createEventModule(publicOnOrderApproved);
export { DescriptionLineType, ItemTypeItemType, PaymentOptionType, JurisdictionType, SubscriptionFrequency, TaxableAddressType, PaymentStatus, FulfillmentStatus, WeightUnit, VatType, PickupMethod, OrderStatus, DiscountType, DiscountReason, LineItemQuantityChangeType, ActivityType, AttributionSource, ChannelType, State, Namespace, SortOrder, TransactionStatus, AuthorizationCaptureStatus, AuthorizationVoidStatus, Reason, ActionType, MembershipPaymentStatus, RefundStatus, WebhookIdentityType, VersioningMode, PreviewEmailType, ScheduledAction, DurationUnit, PaymentCollectabilityStatus, RefundableStatus, NonRefundableReason, ManuallyRefundableReason, RestockType, OrderApprovalStrategy, DeltaPaymentOptionType, InventoryAction, Placement, SubdivisionType, SourceType, CustomFieldGroup, ValueType, DepositType, InvoiceStatus, } from './ecom-v1-order-orders.universal';
