// @wix/identity_site-invite/build/cjs/src/identity-invites-v1-site-invite-site-invite.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.revokeInvite = exports.resendInvite = exports.bulkInvite = void 0;
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressInvitesApiSiteInvitesApiUrl(opts) {
    const domainToMappings = {
        'www._base_domain_': [
            {
                srcPath: '/_api/invites-web',
                destPath: '',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/invites-web',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/invites-web',
                destPath: '',
            },
            {
                srcPath: '/invites',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/invites-web',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/invites-web',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/invites-web',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/invites/invites/site',
                destPath: '/invites/site',
            },
            {
                srcPath: '/invites/invites/account',
                destPath: '/invites/account',
            },
            {
                srcPath: '/invites',
                destPath: '',
            },
            {
                srcPath: '/invites/account-invite',
                destPath: '/invites/account-invite',
            },
            {
                srcPath: '/invites/site-invite',
                destPath: '/site-invite',
            },
        ],
        'dev._base_domain_': [
            {
                srcPath: '/_api/invites-web',
                destPath: '',
            },
        ],
        'platform.rise.ai': [
            {
                srcPath: '/invites/site',
                destPath: '/invites/site',
            },
            {
                srcPath: '/site-invite',
                destPath: '/site-invite',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/identity_site-invite';
/**
 * Creates and sends emails inviting potential site contributors to become contributors in the requesting site.
 * > **Important**: This call requires an account level API key and cannot be authenticated with the standard authorization header. API keys are currently available to selected beta users only.
 */
function bulkInvite(payload) {
    function __bulkInvite({ host }) {
        const metadata = {
            entityFqdn: 'wix.identity.invites.v1.site_invite',
            method: 'POST',
            methodFqn: 'com.wixpress.invites.api.SiteInvitesApi.BulkInvite',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressInvitesApiSiteInvitesApiUrl({
                protoPath: '/invites/site-invite/bulk',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'invites.dateCreated' },
                        { path: 'invites.expirationDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkInvite;
}
exports.bulkInvite = bulkInvite;
/**
 * Resends the email invitation to a potential site contributor.
 * > **Important**: This call requires an account level API key and cannot be authenticated with the standard authorization header. API keys are currently available to selected beta users only.
 */
function resendInvite(payload) {
    function __resendInvite({ host }) {
        const metadata = {
            entityFqdn: 'wix.identity.invites.v1.site_invite',
            method: 'POST',
            methodFqn: 'com.wixpress.invites.api.SiteInvitesApi.ResendInvite',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressInvitesApiSiteInvitesApiUrl({
                protoPath: '/invites/site-invite/{inviteId}/resend',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'invite.dateCreated' },
                        { path: 'invite.expirationDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __resendInvite;
}
exports.resendInvite = resendInvite;
/**
 * Revokes a pending site contributor invite.
 * > **Important**: This call requires an account level API key and cannot be authenticated with the standard authorization header. API keys are currently available to selected beta users only.
 */
function revokeInvite(payload) {
    function __revokeInvite({ host }) {
        const metadata = {
            entityFqdn: 'wix.identity.invites.v1.site_invite',
            method: 'POST',
            methodFqn: 'com.wixpress.invites.api.SiteInvitesApi.RevokeInvite',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressInvitesApiSiteInvitesApiUrl({
                protoPath: '/invites/site-invite/{inviteId}/revoke',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __revokeInvite;
}
exports.revokeInvite = revokeInvite;
