// @wix/ecom_delivery-profile/build/es/src/ecom-v1-delivery-profile-delivery-profile.context.js
import { createDeliveryProfile as publicCreateDeliveryProfile, getDeliveryProfile as publicGetDeliveryProfile, updateDeliveryProfile as publicUpdateDeliveryProfile, deleteDeliveryProfile as publicDeleteDeliveryProfile, queryDeliveryProfiles as publicQueryDeliveryProfiles, addDeliveryRegion as publicAddDeliveryRegion, updateDeliveryRegion as publicUpdateDeliveryRegion, removeDeliveryRegion as publicRemoveDeliveryRegion, addDeliveryCarrier as publicAddDeliveryCarrier, removeDeliveryCarrier as publicRemoveDeliveryCarrier, updateDeliveryCarrier as publicUpdateDeliveryCarrier, listDeliveryCarrierDetails as publicListDeliveryCarrierDetails, listDeliveryCarriers as publicListDeliveryCarriers, updateExtendedFields as publicUpdateExtendedFields, } from './ecom-v1-delivery-profile-delivery-profile.public';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onDeliveryProfileCreated as publicOnDeliveryProfileCreated } from './ecom-v1-delivery-profile-delivery-profile.public';
import { onDeliveryProfileDeliveryRegionAdded as publicOnDeliveryProfileDeliveryRegionAdded } from './ecom-v1-delivery-profile-delivery-profile.public';
import { onDeliveryProfileUpdated as publicOnDeliveryProfileUpdated } from './ecom-v1-delivery-profile-delivery-profile.public';
import { onDeliveryProfileDeleted as publicOnDeliveryProfileDeleted } from './ecom-v1-delivery-profile-delivery-profile.public';
import { onDeliveryProfileDeliveryRegionRemoved as publicOnDeliveryProfileDeliveryRegionRemoved } from './ecom-v1-delivery-profile-delivery-profile.public';
import { onDeliveryProfileDeliveryRegionUpdated as publicOnDeliveryProfileDeliveryRegionUpdated } from './ecom-v1-delivery-profile-delivery-profile.public';
export const createDeliveryProfile = /*#__PURE__*/ createRESTModule(publicCreateDeliveryProfile);
export const getDeliveryProfile = /*#__PURE__*/ createRESTModule(publicGetDeliveryProfile);
export const updateDeliveryProfile = /*#__PURE__*/ createRESTModule(publicUpdateDeliveryProfile);
export const deleteDeliveryProfile = /*#__PURE__*/ createRESTModule(publicDeleteDeliveryProfile);
export const queryDeliveryProfiles = /*#__PURE__*/ createRESTModule(publicQueryDeliveryProfiles);
export const addDeliveryRegion = /*#__PURE__*/ createRESTModule(publicAddDeliveryRegion);
export const updateDeliveryRegion = /*#__PURE__*/ createRESTModule(publicUpdateDeliveryRegion);
export const removeDeliveryRegion = /*#__PURE__*/ createRESTModule(publicRemoveDeliveryRegion);
export const addDeliveryCarrier = /*#__PURE__*/ createRESTModule(publicAddDeliveryCarrier);
export const removeDeliveryCarrier = /*#__PURE__*/ createRESTModule(publicRemoveDeliveryCarrier);
export const updateDeliveryCarrier = /*#__PURE__*/ createRESTModule(publicUpdateDeliveryCarrier);
export const listDeliveryCarrierDetails = /*#__PURE__*/ createRESTModule(publicListDeliveryCarrierDetails);
export const listDeliveryCarriers = /*#__PURE__*/ createRESTModule(publicListDeliveryCarriers);
export const updateExtendedFields = /*#__PURE__*/ createRESTModule(publicUpdateExtendedFields);
export { publicOnDeliveryProfileCreated };
/** */
export const onDeliveryProfileCreated = createEventModule(publicOnDeliveryProfileCreated);
export { publicOnDeliveryProfileDeliveryRegionAdded };
/**
 * Triggered when a delivery_region is added to a delivery_profile.
 */
export const onDeliveryProfileDeliveryRegionAdded = createEventModule(publicOnDeliveryProfileDeliveryRegionAdded);
export { publicOnDeliveryProfileUpdated };
/**
 * Triggered when the delivery_carrier updated successfully
 */
export const onDeliveryProfileUpdated = createEventModule(publicOnDeliveryProfileUpdated);
export { publicOnDeliveryProfileDeleted };
/** */
export const onDeliveryProfileDeleted = createEventModule(publicOnDeliveryProfileDeleted);
export { publicOnDeliveryProfileDeliveryRegionRemoved };
/**
 * Triggered for each delivery_region assigned to the deleted delivery_profile
 */
export const onDeliveryProfileDeliveryRegionRemoved = createEventModule(publicOnDeliveryProfileDeliveryRegionRemoved);
export { publicOnDeliveryProfileDeliveryRegionUpdated };
/**
 * Triggered when the delivery_carrier updated successfully
 */
export const onDeliveryProfileDeliveryRegionUpdated = createEventModule(publicOnDeliveryProfileDeliveryRegionUpdated);
export { ChargeType, SortOrder, State, SiteCreatedContext, Namespace, DeleteStatus, WebhookIdentityType, } from './ecom-v1-delivery-profile-delivery-profile.universal';
