// @wix/ecom_delivery-profile/build/es/src/ecom-v1-delivery-profile-delivery-profile.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { addDeliveryCarrier as universalAddDeliveryCarrier, addDeliveryRegion as universalAddDeliveryRegion, createDeliveryProfile as universalCreateDeliveryProfile, deleteDeliveryProfile as universalDeleteDeliveryProfile, getDeliveryProfile as universalGetDeliveryProfile, listDeliveryCarrierDetails as universalListDeliveryCarrierDetails, listDeliveryCarriers as universalListDeliveryCarriers, queryDeliveryProfiles as universalQueryDeliveryProfiles, removeDeliveryCarrier as universalRemoveDeliveryCarrier, removeDeliveryRegion as universalRemoveDeliveryRegion, updateDeliveryCarrier as universalUpdateDeliveryCarrier, updateDeliveryProfile as universalUpdateDeliveryProfile, updateDeliveryRegion as universalUpdateDeliveryRegion, updateExtendedFields as universalUpdateExtendedFields, } from './ecom-v1-delivery-profile-delivery-profile.universal';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function createDeliveryProfile(httpClient) {
    return (deliveryProfile) => universalCreateDeliveryProfile(deliveryProfile, 
    // @ts-ignore
    { httpClient });
}
export function getDeliveryProfile(httpClient) {
    return (deliveryProfileId) => universalGetDeliveryProfile(deliveryProfileId, 
    // @ts-ignore
    { httpClient });
}
export function updateDeliveryProfile(httpClient) {
    return (_id, deliveryProfile) => universalUpdateDeliveryProfile(_id, deliveryProfile, 
    // @ts-ignore
    { httpClient });
}
export function deleteDeliveryProfile(httpClient) {
    return (deliveryProfileId) => universalDeleteDeliveryProfile(deliveryProfileId, 
    // @ts-ignore
    { httpClient });
}
export function queryDeliveryProfiles(httpClient) {
    return () => universalQueryDeliveryProfiles(
    // @ts-ignore
    { httpClient });
}
export function addDeliveryRegion(httpClient) {
    return (deliveryProfileId, deliveryRegion, options) => universalAddDeliveryRegion(deliveryProfileId, deliveryRegion, options, 
    // @ts-ignore
    { httpClient });
}
export function updateDeliveryRegion(httpClient) {
    return (identifiers, deliveryRegion, options) => universalUpdateDeliveryRegion(identifiers, deliveryRegion, options, 
    // @ts-ignore
    { httpClient });
}
export function removeDeliveryRegion(httpClient) {
    return (identifiers, options) => universalRemoveDeliveryRegion(identifiers, options, 
    // @ts-ignore
    { httpClient });
}
export function addDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => universalAddDeliveryCarrier(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function removeDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => universalRemoveDeliveryCarrier(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateDeliveryCarrier(httpClient) {
    return (deliveryRegionId, options) => universalUpdateDeliveryCarrier(deliveryRegionId, options, 
    // @ts-ignore
    { httpClient });
}
export function listDeliveryCarrierDetails(httpClient) {
    return () => universalListDeliveryCarrierDetails(
    // @ts-ignore
    { httpClient });
}
export function listDeliveryCarriers(httpClient) {
    return (deliveryProfileId, options) => universalListDeliveryCarriers(deliveryProfileId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => universalUpdateExtendedFields(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
export const onDeliveryProfileCreated = EventDefinition('wix.ecom.v1.delivery_profile_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.deliveryRegions.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onDeliveryProfileDeliveryRegionAdded = EventDefinition('wix.ecom.v1.delivery_profile_delivery_region_added', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.createdDate' }, { path: 'metadata.eventTime' }],
    },
])))();
export const onDeliveryProfileUpdated = EventDefinition('wix.ecom.v1.delivery_profile_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.deliveryRegions.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onDeliveryProfileDeleted = EventDefinition('wix.ecom.v1.delivery_profile_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.deliveryRegions.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onDeliveryProfileDeliveryRegionRemoved = EventDefinition('wix.ecom.v1.delivery_profile_delivery_region_removed', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
])))();
export const onDeliveryProfileDeliveryRegionUpdated = EventDefinition('wix.ecom.v1.delivery_profile_delivery_region_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'data.createdDate' }, { path: 'metadata.eventTime' }],
    },
])))();
export { ChargeType, DeleteStatus, Namespace, SiteCreatedContext, SortOrder, State, WebhookIdentityType, } from './ecom-v1-delivery-profile-delivery-profile.universal';
