// @wix/ecom_discount-rules/build/es/src/ecom-discounts-v1-discount-rule-discount-rules.context.js
import { createDiscountRule as publicCreateDiscountRule, getDiscountRule as publicGetDiscountRule, updateDiscountRule as publicUpdateDiscountRule, deleteDiscountRule as publicDeleteDiscountRule, queryDiscountRules as publicQueryDiscountRules, } from './ecom-discounts-v1-discount-rule-discount-rules.public';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onDiscountRuleCreated as publicOnDiscountRuleCreated } from './ecom-discounts-v1-discount-rule-discount-rules.public';
import { onDiscountRuleUpdated as publicOnDiscountRuleUpdated } from './ecom-discounts-v1-discount-rule-discount-rules.public';
import { onDiscountRuleDeleted as publicOnDiscountRuleDeleted } from './ecom-discounts-v1-discount-rule-discount-rules.public';
export const createDiscountRule = /*#__PURE__*/ createRESTModule(publicCreateDiscountRule);
export const getDiscountRule = /*#__PURE__*/ createRESTModule(publicGetDiscountRule);
export const updateDiscountRule = /*#__PURE__*/ createRESTModule(publicUpdateDiscountRule);
export const deleteDiscountRule = /*#__PURE__*/ createRESTModule(publicDeleteDiscountRule);
export const queryDiscountRules = /*#__PURE__*/ createRESTModule(publicQueryDiscountRules);
export { publicOnDiscountRuleCreated };
/**
 * Triggered when a discount rule is created.
 */
export const onDiscountRuleCreated = createEventModule(publicOnDiscountRuleCreated);
export { publicOnDiscountRuleUpdated };
/**
 * Triggered when a discount rule is updated.
 */
export const onDiscountRuleUpdated = createEventModule(publicOnDiscountRuleUpdated);
export { publicOnDiscountRuleDeleted };
/**
 * Triggered when a discount rule is deleted.
 */
export const onDiscountRuleDeleted = createEventModule(publicOnDiscountRuleDeleted);
export { ScopeType, TriggerType, Type, DiscountType, Status, AppliedSubjectType, SortOrder, WebhookIdentityType, } from './ecom-discounts-v1-discount-rule-discount-rules.universal';
