// @wix/ecom_current-cart/build/es/src/ecom-v1-cart-current-cart.context.js
import { getCurrentCart as publicGetCurrentCart, updateCurrentCart as publicUpdateCurrentCart, addToCurrentCart as publicAddToCurrentCart, removeLineItemsFromCurrentCart as publicRemoveLineItemsFromCurrentCart, createCheckoutFromCurrentCart as publicCreateCheckoutFromCurrentCart, removeCouponFromCurrentCart as publicRemoveCouponFromCurrentCart, updateCurrentCartLineItemQuantity as publicUpdateCurrentCartLineItemQuantity, estimateCurrentCartTotals as publicEstimateCurrentCartTotals, deleteCurrentCart as publicDeleteCurrentCart, } from './ecom-v1-cart-current-cart.public';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCartUpdated as publicOnCartUpdated } from './ecom-v1-cart-current-cart.public';
import { onCartDeleted as publicOnCartDeleted } from './ecom-v1-cart-current-cart.public';
import { onCartCreated as publicOnCartCreated } from './ecom-v1-cart-current-cart.public';
export const getCurrentCart = /*#__PURE__*/ createRESTModule(publicGetCurrentCart);
export const updateCurrentCart = /*#__PURE__*/ createRESTModule(publicUpdateCurrentCart);
export const addToCurrentCart = /*#__PURE__*/ createRESTModule(publicAddToCurrentCart);
export const removeLineItemsFromCurrentCart = /*#__PURE__*/ createRESTModule(publicRemoveLineItemsFromCurrentCart);
export const createCheckoutFromCurrentCart = /*#__PURE__*/ createRESTModule(publicCreateCheckoutFromCurrentCart);
export const removeCouponFromCurrentCart = /*#__PURE__*/ createRESTModule(publicRemoveCouponFromCurrentCart);
export const updateCurrentCartLineItemQuantity = /*#__PURE__*/ createRESTModule(publicUpdateCurrentCartLineItemQuantity);
export const estimateCurrentCartTotals = /*#__PURE__*/ createRESTModule(publicEstimateCurrentCartTotals);
export const deleteCurrentCart = /*#__PURE__*/ createRESTModule(publicDeleteCurrentCart);
export { publicOnCartUpdated };
/**
 * Triggered when a cart is updated.
 */
export const onCartUpdated = createEventModule(publicOnCartUpdated);
export { publicOnCartDeleted };
/**
 * Triggered when a cart is deleted.
 */
export const onCartDeleted = createEventModule(publicOnCartDeleted);
export { publicOnCartCreated };
/**
 * Triggered when a cart is created.
 */
export const onCartCreated = createEventModule(publicOnCartCreated);
export { DescriptionLineType, ItemAvailabilityStatus, ItemTypeItemType, SubscriptionFrequency, FileType, PaymentOptionType, TaxableAddressType, WeightUnit, VatType, JurisdictionType, RateType, ManualCalculationReason, FallbackReason, PickupMethod, ChargeType, DiscountType, RuleType, Severity, NameInOther, NameInLineItem, SuggestedFix, ChannelType, WebhookIdentityType, } from './ecom-v1-cart-current-cart.universal';
