// @wix/ecom_tips/build/es/src/tips-v1-tip-tips.types.js
export var IdentityType;
(function (IdentityType) {
    IdentityType["UNKNOWN"] = "UNKNOWN";
    IdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    IdentityType["MEMBER"] = "MEMBER";
    IdentityType["WIX_USER"] = "WIX_USER";
    IdentityType["APP"] = "APP";
})(IdentityType || (IdentityType = {}));
export var TipType;
(function (TipType) {
    /** There is no information about the tip type. */
    TipType["UNKNOWN_TIP_TYPE"] = "UNKNOWN_TIP_TYPE";
    /** The tip is calculated as a percentage of the subtotal of associated line items. */
    TipType["PERCENTAGE"] = "PERCENTAGE";
    /** The tip is a fixed value, pre-configured from the relevant settings presets. */
    TipType["AMOUNT"] = "AMOUNT";
    /** The customer, or the business on behalf of the customer, manually set a tip amount, which may not match any preset values. */
    TipType["CUSTOM"] = "CUSTOM";
})(TipType || (TipType = {}));
export var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
export var QueryBy;
(function (QueryBy) {
    QueryBy["UNKNOWN_QUERY_BY"] = "UNKNOWN_QUERY_BY";
    QueryBy["ORDER"] = "ORDER";
    QueryBy["TIP"] = "TIP";
})(QueryBy || (QueryBy = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
