// @wix/ecom_memberships/build/cjs/src/service-plugins-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityType = exports.MembershipCannotBeChargedType = exports.MembershipErrorType = void 0;
var MembershipErrorType;
(function (MembershipErrorType) {
    MembershipErrorType["UNKNOWN"] = "UNKNOWN";
    MembershipErrorType["MEMBERSHIP_NOT_FOUND"] = "MEMBERSHIP_NOT_FOUND";
    /** which line item ids */
    MembershipErrorType["LINE_ITEM_NOT_FOUND"] = "LINE_ITEM_NOT_FOUND";
    /** which ids and why */
    MembershipErrorType["MEMBERSHIP_CANNOT_BE_USED_FOR_LINE_ITEMS"] = "MEMBERSHIP_CANNOT_BE_USED_FOR_LINE_ITEMS";
    /** reason */
    MembershipErrorType["MEMBERSHIP_SELECTION_INVALID"] = "MEMBERSHIP_SELECTION_INVALID";
})(MembershipErrorType || (exports.MembershipErrorType = MembershipErrorType = {}));
var MembershipCannotBeChargedType;
(function (MembershipCannotBeChargedType) {
    MembershipCannotBeChargedType["OUT_OF_CREDITS"] = "OUT_OF_CREDITS";
    MembershipCannotBeChargedType["NOT_STARTED_YET"] = "NOT_STARTED_YET";
    MembershipCannotBeChargedType["ENDED"] = "ENDED";
    MembershipCannotBeChargedType["NOT_APPLICABLE_FOR_MULTIPLE_PARTICIPANTS"] = "NOT_APPLICABLE_FOR_MULTIPLE_PARTICIPANTS";
})(MembershipCannotBeChargedType || (exports.MembershipCannotBeChargedType = MembershipCannotBeChargedType = {}));
var IdentityType;
(function (IdentityType) {
    IdentityType["UNKNOWN"] = "UNKNOWN";
    IdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    IdentityType["MEMBER"] = "MEMBER";
    IdentityType["WIX_USER"] = "WIX_USER";
    IdentityType["APP"] = "APP";
})(IdentityType || (exports.IdentityType = IdentityType = {}));
