// @wix/identity_site-invite/build/es/src/identity-invites-v1-site-invite-site-invite.meta.js
import * as ambassadorWixIdentityInvitesV1SiteInvite from './identity-invites-v1-site-invite-site-invite.http';
export function bulkInvite() {
    const payload = {};
    const getRequestOptions = ambassadorWixIdentityInvitesV1SiteInvite.bulkInvite(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/invites/site-invite/bulk',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function resendInvite() {
    const payload = { inviteId: ':inviteId' };
    const getRequestOptions = ambassadorWixIdentityInvitesV1SiteInvite.resendInvite(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/invites/site-invite/{inviteId}/resend',
        pathParams: { inviteId: 'inviteId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function revokeInvite() {
    const payload = { inviteId: ':inviteId' };
    const getRequestOptions = ambassadorWixIdentityInvitesV1SiteInvite.revokeInvite(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/invites/site-invite/{inviteId}/revoke',
        pathParams: { inviteId: 'inviteId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
