// @wix/identity_roles-management/build/cjs/src/identity-v1-contributor-roles-management.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.querySiteContributors = exports.changeRole = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressRolesManagementApiSiteRolesManagementServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/contributors/accept',
                destPath: '/contributors/accept',
            },
            {
                srcPath: '/roles-management-web',
                destPath: '',
            },
            {
                srcPath: '/team/accept',
                destPath: '/team/accept',
            },
            {
                srcPath: '/_api/roles-management-web',
                destPath: '',
            },
            {
                srcPath: '/app/accept',
                destPath: '/app/accept',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/roles-management-web',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/roles-management-web',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/roles-management-web',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/roles-management-web',
                destPath: '',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/roles-management-web',
                destPath: '',
            },
            {
                srcPath: '/roles-management-web',
                destPath: '',
            },
            {
                srcPath: '/contributors/accept',
                destPath: '/contributors/accept',
            },
            {
                srcPath: '/team/accept',
                destPath: '/team/accept',
            },
        ],
        'dev._base_domain_': [
            {
                srcPath: '/_api/roles-management-web',
                destPath: '',
            },
        ],
        'platform.rise.ai': [
            {
                srcPath: '/v2/contributors',
                destPath: '/v2/contributors',
            },
            {
                srcPath: '/contributor/change/role',
                destPath: '/contributor/change/role',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/roles-management',
                destPath: '',
            },
            {
                srcPath: '/roles-management-web',
                destPath: '',
            },
            {
                srcPath: '/roles-management/roles',
                destPath: '/roles',
            },
            {
                srcPath: '/roles-management/team',
                destPath: '/team',
            },
            {
                srcPath: '/roles-management/v3/team',
                destPath: '/v3/team',
            },
            {
                srcPath: '/roles-management/v2/team',
                destPath: '/v2/team',
            },
            {
                srcPath: '/roles-management/contributor/change',
                destPath: '/contributor/change',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/identity_roles-management';
/** Overrides all the roles of a contributor for the specified site. */
function changeRole(payload) {
    function __changeRole({ host }) {
        const metadata = {
            entityFqdn: 'wix.identity.v1.contributor',
            method: 'PUT',
            methodFqn: 'com.wixpress.roles.management.api.SiteRolesManagementService.ChangeRole',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressRolesManagementApiSiteRolesManagementServiceUrl({
                protoPath: '/roles-management/contributor/change/role',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __changeRole;
}
exports.changeRole = changeRole;
/** Retrieves a list of contributors for the specified site, given the provided filters. */
function querySiteContributors(payload) {
    function __querySiteContributors({ host }) {
        const metadata = {
            entityFqdn: 'wix.identity.v1.contributor',
            method: 'GET',
            methodFqn: 'com.wixpress.roles.management.api.SiteRolesManagementService.QuerySiteContributors',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressRolesManagementApiSiteRolesManagementServiceUrl({
                protoPath: '/v2/contributors/query',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __querySiteContributors;
}
exports.querySiteContributors = querySiteContributors;
