// @wix/identity_roles-management/build/cjs/src/identity-v1-contributor-roles-management.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.SubjectType = exports.SubjectContextType = exports.ResourceType = exports.InviteStatus = exports.FieldSet = exports.ConditionAttributeType = exports.querySiteContributors = exports.changeRole = exports.__metadata = void 0;
const identity_v1_contributor_roles_management_universal_1 = require("./identity-v1-contributor-roles-management.universal");
exports.__metadata = { PACKAGE_NAME: '@wix/identity' };
function changeRole(httpClient) {
    return (accountId, options) => (0, identity_v1_contributor_roles_management_universal_1.changeRole)(accountId, options, 
    // @ts-ignore
    { httpClient });
}
exports.changeRole = changeRole;
function querySiteContributors(httpClient) {
    return (options) => (0, identity_v1_contributor_roles_management_universal_1.querySiteContributors)(options, 
    // @ts-ignore
    { httpClient });
}
exports.querySiteContributors = querySiteContributors;
var identity_v1_contributor_roles_management_universal_2 = require("./identity-v1-contributor-roles-management.universal");
Object.defineProperty(exports, "ConditionAttributeType", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.ConditionAttributeType; } });
Object.defineProperty(exports, "FieldSet", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.FieldSet; } });
Object.defineProperty(exports, "InviteStatus", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.InviteStatus; } });
Object.defineProperty(exports, "ResourceType", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.ResourceType; } });
Object.defineProperty(exports, "SubjectContextType", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.SubjectContextType; } });
Object.defineProperty(exports, "SubjectType", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.SubjectType; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return identity_v1_contributor_roles_management_universal_2.Type; } });
