// @wix/identity_roles-management/build/cjs/src/identity-v1-contributor-roles-management.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldSet = exports.Type = exports.InviteStatus = exports.SubjectContextType = exports.SubjectType = exports.ConditionAttributeType = exports.ResourceType = void 0;
var ResourceType;
(function (ResourceType) {
    ResourceType["UNKNOWN_RESOURCE_TYPE"] = "UNKNOWN_RESOURCE_TYPE";
    ResourceType["SITE"] = "SITE";
})(ResourceType || (exports.ResourceType = ResourceType = {}));
var ConditionAttributeType;
(function (ConditionAttributeType) {
    ConditionAttributeType["UNKNOWN_CONDITION_TYPE"] = "UNKNOWN_CONDITION_TYPE";
    ConditionAttributeType["FOLDER"] = "FOLDER";
})(ConditionAttributeType || (exports.ConditionAttributeType = ConditionAttributeType = {}));
var SubjectType;
(function (SubjectType) {
    SubjectType["UNKNOWN"] = "UNKNOWN";
    SubjectType["ACCOUNT"] = "ACCOUNT";
    SubjectType["USER"] = "USER";
    SubjectType["USER_GROUP"] = "USER_GROUP";
    SubjectType["MEMBER_GROUP"] = "MEMBER_GROUP";
    SubjectType["VISITOR_GROUP"] = "VISITOR_GROUP";
    SubjectType["EXTERNAL_APP"] = "EXTERNAL_APP";
    SubjectType["ACCOUNT_GROUP"] = "ACCOUNT_GROUP";
    SubjectType["WIX_APP"] = "WIX_APP";
})(SubjectType || (exports.SubjectType = SubjectType = {}));
var SubjectContextType;
(function (SubjectContextType) {
    SubjectContextType["UNKNOWN_CTX"] = "UNKNOWN_CTX";
    SubjectContextType["ORG_CTX"] = "ORG_CTX";
    SubjectContextType["ACCOUNT_CTX"] = "ACCOUNT_CTX";
})(SubjectContextType || (exports.SubjectContextType = SubjectContextType = {}));
/** Invite status stating whether the invite was accepted, waiting to be accepted, deleted etc.. */
var InviteStatus;
(function (InviteStatus) {
    InviteStatus["Pending"] = "Pending";
    InviteStatus["Used"] = "Used";
    InviteStatus["Deleted"] = "Deleted";
    InviteStatus["Declined"] = "Declined";
    InviteStatus["Expired"] = "Expired";
})(InviteStatus || (exports.InviteStatus = InviteStatus = {}));
/** Enum to represent different types of contributors quota. */
var Type;
(function (Type) {
    Type["UNKNOWN"] = "UNKNOWN";
    Type["LIMITED"] = "LIMITED";
    Type["UNLIMITED"] = "UNLIMITED";
})(Type || (exports.Type = Type = {}));
var FieldSet;
(function (FieldSet) {
    FieldSet["UNKNOWN"] = "UNKNOWN";
    /** Include only `account_id` and `account_owner_id` fields. */
    FieldSet["META_DATA"] = "META_DATA";
})(FieldSet || (exports.FieldSet = FieldSet = {}));
